src/components/EditDonationModal/EditDonationModal.tsx

```diff
@@ -248,10 +261,19 @@ export const EditDonationModal: React.FC<EditDonationModalProps> = ({
                         name: donation.donorAccount.displayName,
                       },
                     ]}
-                    onChange={(donorAccountId) =>
-                      setFieldValue('donorAccountId', donorAccountId)
-                    }
+                    onChange={(donorAccountId) => {
+                      setFieldValue('donorAccountId', donorAccountId);
+                      setFieldTouched('donorAccountId', true, false);
+                    }}
+                    onBlur={() => {
+                      setFieldTouched('donorAccountId', true);
```

Comment:
Could you also please add a test to ensure your code resolves the inconsistency you're addressing?